libpyvinyl/Instrument.py

```diff
@@ -162,3 +169,74 @@ def __str__(self) -> str:
         mystring += repr(self.parameters)
         mystring += f"############"
         return mystring
+
+    @property
+    def samples(self) -> List[str]:
+        """Return the list of implemented samples"""
+        return self.__implemented_samples
+
+    @samples.setter
+    def samples(self, l: List[str]) -> None:
+        """Set the list of implemented samples (to be used in the init of the derived class"""
+        self.__implemented_samples = l
+
+    @property
+    def sample(self) -> Any:
+        """Return the object representing the sample"""
+        return self.__sample
+
+    @sample.setter
+    def sample(self, obj: Any) -> None:
+        """Set the object implementing the sample"""
+        self.__sample = obj
+
+    @property
+    def sample_name(self) -> str:
+        """Return the name of the currently implemented sample"""
+        return self.__sample_name
+
+    @sample_name.setter
+    def sample_name(self, name: str) -> None:
+        """Set the name of currently implemented sample. To be performed by the *set_sample_by_name(name)* method"""
+        self.__sample_name = name
+
+    def set_sample_by_name(self, name: str) -> None:
+        """Set the sample component by name: to be implemented in the derived class"""
+        # This setter should be implemented in the specialized class
+        raise NotImplementedError(
+            self.__class__.__name__
+            + " sample setter not implemented in the specialized Instrument class"
+        )
+
+    @property
+    def sample_environments(self) -> List[str]:
+        """Return the list of implemented sample environments"""
+        return self.__implemented_sample_environments
+
+    @sample_environments.setter
+    def sample_environments(self, l: List[str]) -> None:
+        """ """
+        self.__implemented_sample_environments = l
+
+    @property
+    def sample_environment(self) -> Any:
+        """Return the object representing the currently used sample environment"""
+        return self.__sample_environment
+
+    @sample_environment.setter
+    def sample_environment(self, obj: Any) -> None:
+        """Set the object implementing the sample environment"""
+        self.__sample_environment = obj
+
+    @property
+    def sample_environment_name(self):
+        """Return the name of the currently implemented sample environment"""
+        return self.__sample_environment_name
+
+    def set_sample_environment_by_name(self, name: str) -> None:
+        """Setting the sample environment name"""
+        # This setter should be implemented in the specialized class
+        raise NotImplementedError(
+            self.__class__.__name__
+            + " sample setter not implemented in the specialized Instrument class"
+        )
```

Comment:
Should `sample_environment.setter` call `set_sample_environment_by_name`? Otherwise the `NotImplementedError` in `test_instrument.py` will never be raised.